jenkins/opensearch/integ-test.jenkinsfile

```diff
@@ -0,0 +1,109 @@
+lib = library(identifier: "jenkins@20211118", retriever: legacySCM(scm))
+
+pipeline {
+    agent none
+    environment {
+        AWS_ROLE_ARN = "arn:aws:iam::${AWS_ACCOUNT_PUBLIC}:role/opensearch-test"
+        AWS_ROLE_SESSION_NAME = "jenkins-test-session"
+    }
+    tools {
+        jdk "JDK14"
+        maven "maven-3.8.2"
+    }
+    parameters {
+        string(
+            defaultValue: '',
+            name: 'TEST_MANIFEST',
+            description: 'Test manifest under the manifests folder, e.g. manifests/2.0.0/opensearch-2.0.0-test.yml.',
+            trim: true
+        )
+        string(
+            defaultValue: '',
+            name: 'BUILD_MANIFEST_URL',
+            description: 'The build manifest URL, e.g. https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/1.2.2/98/linux/x64/builds/opensearch/manifest.yml.',
+            trim: true
+        )
+        string(
+            defaultValue: '',
+            name: 'AGENT_LABEL',
+            description: 'The agent label where the tests should be executed, e.g. Jenkins-Agent-al2-x64-m5xlarge.',
+            trim: true
+        )
+    }
+    stages {
+        stage('integ-test') {
+            agent {
+                node {
+                    label "${AGENT_LABEL}"
+                }
+            }
+            steps {
+                script {
+                    // Fetching the build-specific values and assigning to their env vars
+                    (buildManifestPath, buildId) = getBuildManifestInfo()
```

Comment:
This API is a bit strange, I'd prefer it'd download a manifest and return that.

```
def buildManifest = downloadBuildManifest(manifest: "${BUILD_MANIFEST_URL}")
...
```

Reply:
I much prefer this approach.  Functions with no side effects are my jam

Reply:
Do you mean as implemented now? 

Note that shared functions are global. This one does not "get build manifest info", it "get some very specific build manifest info used in integ-test workflow", are we going to be adding another one like these when we need some other value from a remote manifest?

Reply:
I've updated this, see https://github.com/opensearch-project/opensearch-build/pull/1429#discussion_r786970899